### PR TITLE
replace reviewer with autofix and verify-conversation skills

### PR DESCRIPTION
**Pull request by evgunter**
replace reviewer with autofix + separate verify conversation skill. these are both copied from https://github.com/imbue-ai/hammer-verify, which also includes the script used to generate them from vet, which i can also copy over if desired

need to check if the "mark which parts of the conversation have already been reviewed" logic works 

---

**Note:** The `model: opus[1m]` frontmatter field in verify-conversation.md needs manual verification -- the `[1m]` suffix is documented for `/model` and `--model` but it's unclear whether skill frontmatter parses it the same way. Test this before merging.

## Summary

Replace the separate reviewer agent (verify-branch, stop_hook_reviewer.sh, run_reviewer.sh) with two new skills triggered by stop hooks:

- **/autofix** -- Iteratively verifies and fixes code issues using subagents, with each fix in a separate commit. Creates `.autofix/plans/{hash}_verified.md` marker when done. Configurable via `.autofix/config/stop-hook.toml` (enable/disable, extra args).
- **/verify-conversation** -- Analyzes the conversation transcript for behavioral issues (misleading behavior, disobeyed instructions, instructions to save). Creates `.reviews/conversation/{hash}.json` marker when done. Tracks review progress incrementally via `.reviews/conversation/progress.jsonl` so already-reviewed content is skipped on subsequent runs.

### Stop hooks

Both stop hooks are injected via `cli_args` in `.mng/settings.toml` `[agent_types.coder]`, so they only fire for MNG-managed coder agents (not vanilla Claude Code sessions). The hooks check for hash-labeled marker files and block until the corresponding skill has been run.

### Autofix config

`.autofix/config/stop-hook.toml` controls the autofix stop hook:
- `enabled` -- set to `false` to disable the hook entirely
- `extra_args` -- text appended to the `/autofix` command in the hook message (e.g. `"Only fix MAJOR and CRITICAL issues."`)

`.autofix/config/auto-accept.md` defines rules for auto-accepting fixes without user review (now checked into version control instead of symlinked).

### Verify-conversation config and manual invocation

`.reviews/config/verify-conversation.toml` controls which transcript sources are included:
- `include_tracked_sessions` -- sessions from MNG's session history
- `include_current_session` -- the current session via `MNG_CLAUDE_SESSION_ID`
- `include_all_agent_sessions` -- scan the agent's project directory for all session files
- `include_subagents` -- include subagent transcripts alongside their parent sessions

All default to `true`. The skill can also be invoked manually with natural language arguments to narrow scope per-invocation, e.g.:
- `/verify-conversation only review tracked sessions`
- `/verify-conversation skip subagents`
- `/verify-conversation only review the current session`

These override the config file for that run. If total transcript size exceeds 3MB (too large for the 1M context window), the skill warns and suggests narrowing scope.

### Session ID

A `SessionStart` hook (also in `cli_args`) exports `MNG_CLAUDE_SESSION_ID` via `CLAUDE_ENV_FILE`, making the current session ID available to transcript scripts. The transcript discovery scripts (`export_transcript.sh`, `export_transcript_paths.sh`) now search `CLAUDE_CONFIG_DIR` first (for MNG agents with custom config dirs), then fall back to `~/.claude/projects/`. Subagent transcripts at `{session_dir}/subagents/*.jsonl` are also included.

### Other changes

- Simplified `main_claude_stop_hook.sh` to only handle PR/CI (removed reviewer orchestration, stuck-agent tracking, parallel launch logic, dead HOOK_INPUT variable, duplicate error messages)
- Fixed `set -euo pipefail` incompatibility with synchronous PR/CI script execution
- Removed reviewer tmux window from coder template
- Cleaned up related infrastructure (status line issue counting, gitignore entries, dead volume mount)
- Deduplicated session IDs in transcript scripts
- Updated CLAUDE.md to reference /autofix and /verify-conversation

## Test plan

- [x] All 3152 tests pass with 83.58% coverage (threshold: 80%)
- [x] Stop hooks fire correctly and block until skills are run
- [x] Autofix subagents correctly identify and fix issues (tested on this branch -- found and fixed 4 issues across 3 iterations)
- [x] `MNG_CLAUDE_SESSION_ID` is set correctly in new MNG agent sessions (tested with `mng create -t coder`)
- [x] Autofix config (`stop-hook.toml`) correctly controls enable/disable and extra_args message
- [x] Transcript scripts find session files via `CLAUDE_CONFIG_DIR` (was broken before, now fixed)
- [x] Transcript scripts include subagent transcripts with parent source labels
- [ ] Verify verify-conversation produces real review output with available transcript
- [ ] Verify `model: opus[1m]` in skill frontmatter actually selects the 1M context model

### CODE REVIEW

File: .autofix/config/stop-hook.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Configuration for the autofix stop hook.
+# This file is read by scripts/stop_hook_autofix.sh.
+
+# Set to false to disable the autofix stop hook entirely.
+enabled = true
+
+# Extra input appended to the /autofix command in the stop hook message.
+# The agent sees "Run: /autofix <extra_args>" and passes this text to the skill.
+# Examples:
+#   extra_args = "Only fix MAJOR and CRITICAL issues."
```
Comment (evgunter) [inline .autofix/config/stop-hook.toml]: figured this would be more useful than a predefined effort setting

Comment (joshalbrecht) [reply .autofix/config/stop-hook.toml]: uh, hmm, toml turns out not to be a great format for settings, bc it's harder to parse from the command line? how are you loading this...

File: .claude/skills/autofix/validate-diff.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+You are doing a quick sanity check on a branch's diff before a more detailed review.
```
[the remark this answers is not in the document]
Comment (joshalbrecht) [reply .claude/skills/autofix/validate-diff.md]: lol, ok--is there a less strange way to do that? Should they be proper skills? Or stored somewhere else?

File: .claude/skills/autofix/verify-and-fix.md [hunk 3]
```diff
@@ -1,92 +1,77 @@
...
+```
 
-When finished with all issues, touch this file:
+Repeat for each issue. Each fix MUST be its own separate commit.
 
-    .reviews/final_issue_json/!`tmux display-message -t "$TMUX_PANE" -p '#W' || echo reviewer_0`.json.done
+# Step 4: Post-fix Validation
 
-!`rm -rf .reviews/final_issue_json/$(tmux display-message -t "$TMUX_PANE" -p '#W' || echo reviewer_0).json.done`
+After all fixes are committed, run the project test suite. Use whatever test command is specified in the project's CLAUDE.md or README. If none is specified, try `uv run pytest` or the most obvious equivalent.
```
Comment (joshalbrecht) [inline .claude/skills/autofix/verify-and-fix.md]: isn't this going to really slow things down / cause a lot of test churn?

by doing each fix independently, that means that the tests are run N (# of fixes) times instead of I (# of iterations) times...

File: scripts/main_claude_stop_hook.sh [hunk 4]
```diff
@@ -155,130 +151,22 @@ if [[ "$IS_INFORMATIONAL_ONLY" == "true" ]]; then
...
-        if [[ $UNIQUE_COUNT -eq 1 ]]; then
-            echo "ERROR: This hook has been run 3 times at the same commit." >&2
-            echo "ERROR: The agent appears to be stuck and unable to make progress." >&2
-            echo "ERROR: Please investigate and resolve the issue manually." >&2
-            notify_user || echo "No notify_user function defined, skipping."
-            exit 1
-        fi
-    fi
-fi
-
```
Comment (joshalbrecht) [inline scripts/main_claude_stop_hook.sh]: yeah, should probably put this back

File: style_guide.md [hunk 5]
```diff
@@ -1695,6 +1695,16 @@ Tests should be careful to fully isolate themselves. They should be able to run
...
 
+### Test quality
+
+- Every assertion must be able to fail if the code under test has a bug. Tautological assertions (e.g., asserting a constructor field equals the value you just passed in) provide no value.
+- Assert on the *effect* of an operation, not on values you constructed yourself. Test behavior, not implementation details.
+- "No exception raised" is not a sufficient assertion -- verify the actual result.
+- When testing errors, verify the specific error type/message.
+- Use `pytest.mark.parametrize` when multiple cases share identical test logic and differ only in inputs/outputs. Use separate test functions when cases have meaningfully different setup or assertions.
+- Don't write tests just to hit a coverage number. Prioritize: error paths and edge cases > happy-path variations > trivial operations.
+- For collections: test empty, single element, typical, and boundary inputs. For conditionals: test each branch with assertions specific to that branch.
```
[the remark this answers is not in the document]
Comment (joshalbrecht) [reply style_guide.md]: ahh--I believe that Daniel was possibly going to port them over?

also, I see now that this is where these changes moved BUT:
A) we should probably put them in vet anyway, they're just useful
B) I *did* have a few extra strings and updates in there as well (relative to the vet prompts, see the diff on that file)

File: .claude/commands/autofix.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+- If it reports the work looks incomplete, note this but proceed -- autofix works on whatever is there.
+
+### Phase 3: Fix Loop
+
+Create the .autofix/plans directory if it does not already exist.
+
+Repeat up to 10 times:
+
+1. Record the current HEAD as `pre_iteration_head`.
+2. Read the supporting file `.claude/skills/autofix/verify-and-fix.md`. Spawn a single Agent (`subagent_type: "general-purpose"`) with its contents as the prompt. Prepend the line `Base branch for this project: {base_branch}` to the prompt.
```
[the remark this answers is not in the document]
Comment (evgunter) [reply .claude/commands/autofix.md]: - hmm this was supposed to _tell the subagent_ to read the supporting file, but i guess it misunderstood; will fix
- i can make it a slash command but that would mean that random skills we don't actually want to use as humans show up in our slash commands

File: .reviews/config/verify-conversation.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+    "include_tracked_sessions": true,
+    "include_current_session": true,
+    "include_all_agent_sessions": true,
+    "include_subagents": true
```
[the remark this answers is not in the document]
Comment (evgunter) [reply .reviews/config/verify-conversation.json]: i was thinking of making these skills into their own package (right now nothing's really specific to mng so i think it'd be cool to make them publicly available), and so i somewhat wanted to keep their config separate from the other config

File: .claude/commands/autofix.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+Write a brief description of what the branch is trying to do. This helps the diff validation and fix agents distinguish intentional changes from issues.
+
+### Phase 2: Validate the Diff
+
+Read the diff validation prompt from `.claude/skills/autofix/validate-diff.md`. Spawn an Agent (`subagent_type: "general-purpose"`) with that prompt, providing the base branch name and the problem description.
+
+Based on the agent's response:
+- If the diff is empty, STOP and ask the user whether the work has been committed yet or whether the base branch is wrong.
+- If it reports significant unrelated changes, STOP and ask the user what the correct base branch is.
+- If it reports the work looks incomplete, note this but proceed -- autofix works on whatever is there.
```
[the remark this answers is not in the document]
Comment (evgunter) [reply .claude/commands/autofix.md]: in my usage so far these have been effective for catching "oops, diff accidentally includes 50 unrelated commits from main" and not complaining if the base branch is actually correct

File: .mng/settings.toml [hunk 9]
```diff
@@ -15,18 +15,17 @@ target_path = "/code/mng"
 
 [create_templates.coder]
 type = "coder"
-extra_window = ["reviewer_0='export IS_SANDBOX=1 && claude --dangerously-skip-permissions'"]
 
 [create_templates.modal]
 new_host = "modal"
-build_arg = ["file=libs/mng/imbue/mng/resources/Dockerfile", "context-dir=.mng/dev/build/", "volume=code-review-json:/code_reviews"]
```
[the remark this answers is not in the document]
Comment (evgunter) [reply .mng/settings.toml]: it is now slightly tested (but not on modal because i haven't yet debugged why making modal agents doesn't work anymore for me)
